### PR TITLE
#101 Documentation

### PR DESCRIPTION
**Pull request by albamig**
- First version of documentation.

This PR closes issue #101.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # You can set these variables from the command line.
 SPHINXOPTS    =
 SPHINXBUILD   = python -msphinx
-SPHINXPROJ    = minos_microservice_saga
+SPHINXPROJ    = minos_microservice_common
```
Comment (garciparedes) [inline docs/Makefile:7]: I think this would be referred to the `minos-microservice-saga` repo, I am right?

File: docs/conf.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import sys
 sys.path.insert(0, os.path.abspath('..'))
 
-from minos import saga
+from minos import common
```
Comment (garciparedes) [inline docs/conf.py:24]: I think this would be referred to the `minos-microservice-saga` repo, I am right?

File: docs/conf.py [hunk 3]
```diff
@@ -31,33 +31,41 @@
...
-source_suffix = '.rst'
+source_suffix = ['.rst', '.md']
 
 # The master toctree document.
 master_doc = 'index'
 
 # General information about the project.
-project = 'Minos Microservice SAGA'
-copyright = "2021, Clariteia Devs"
+project = 'Minos Microservice Common'
```
Comment (garciparedes) [inline docs/conf.py:57]: I think this would be referred to the `minos-microservice-saga` repo, I am right?

**Review by garciparedes — CHANGES_REQUESTED** [review]
Hi @albamig, the changes are really fine! But, I added you some comments about small bugs related with copy-pasting from `minos-microservice-common`. Could you fix it before I approve the Pull Request? 🤗

File: docs/conf.py [hunk 4]
```diff
@@ -128,7 +147,7 @@
 # [howto, manual, or own class]).
 latex_documents = [
     (master_doc, 'minos.tex',
-     'Minos Microservice SAGA Documentation',
+     'Minos Microservice Common Documentation',
```
Comment (garciparedes) [inline docs/conf.py:150]: Shouldn't this be `Saga`?
```suggestion
     'Minos Microservice Saga Documentation',
```

File: docs/conf.py [hunk 5]
```diff
@@ -139,7 +158,7 @@
 # (source start file, name, description, authors, manual section).
 man_pages = [
     (master_doc, 'minos',
-     'Minos Microservice SAGA Documentation',
+     'Minos Microservice Common Documentation',
```
Comment (garciparedes) [inline docs/conf.py:161]: Shouldn't this be `Saga`?
```suggestion
     'Minos Microservice Saga Documentation',
```

File: docs/conf.py [hunk 6]
```diff
@@ -151,12 +170,33 @@
 #  dir menu entry, description, category)
 texinfo_documents = [
     (master_doc, 'minos',
-     'Minos Microservice SAGA Documentation',
+     'Minos Microservice Common Documentation',
```
Comment (garciparedes) [inline docs/conf.py:173]: Shouldn't this be `Saga`?
```suggestion
     'Minos Microservice Saga Documentation',
```

File: docs/make.bat [hunk 7]
```diff
@@ -9,7 +9,7 @@ if "%SPHINXBUILD%" == "" (
 )
 set SOURCEDIR=.
 set BUILDDIR=_build
-set SPHINXPROJ=minos_microservice_saga
+set SPHINXPROJ=minos_microservice_common
```
Comment (garciparedes) [inline docs/make.bat:12]: Shouldn't this be `minos_microservice_saga`?
```suggestion
set SPHINXPROJ=minos_microservice_saga
```

File: docs/usage.rst [hunk 8]
```diff
@@ -2,6 +2,8 @@
 Usage
 =====
 
-To use Minos Microservice SAGA in a project::
+To use Minos Microservice Common in a project::
+
+    import minos_microservice_common
```
Comment (garciparedes) [inline docs/usage.rst:7]:
```suggestion
    from minos import saga
```

File: docs/conf.py [hunk 9]
```diff
@@ -151,12 +170,33 @@
...
 
+apidoc_toc_file = False
+apidoc_module_first = True
+apidoc_extra_args = [
+    "--force",
+    "--implicit-namespaces",
+]
+apidoc_excluded_paths = [
+    '../minos/common/testing.py'
+]
```
Comment (garciparedes) [inline docs/conf.py:198]: This is not needed here.
```suggestion
```

File: README.md [hunk 10]
```diff
@@ -1,15 +1,79 @@
-Minos Microservice Saga
-=======================
+# Minos Microservice Saga
 
-[![codecov](https://codecov.io/gh/Clariteia/minos_microservice_saga/branch/main/graph/badge.svg)](https://codecov.io/gh/Clariteia/minos_microservice_saga)
+[![codecov](https://codecov.io/gh/Clariteia/minos_microservice_common/branch/main/graph/badge.svg)](https://codecov.io/gh/Clariteia/minos_microservice_common)
+![Tests](https://github.com/Clariteia/minos_microservice_common/actions/workflows/python-tests.yml/badge.svg)
```
Comment (garciparedes) [inline README.md:4]:
```suggestion
[![codecov](https://codecov.io/gh/Clariteia/minos_microservice_saga/branch/main/graph/badge.svg)](https://codecov.io/gh/Clariteia/minos_microservice_saga)
![Tests](https://github.com/Clariteia/minos_microservice_saga/actions/workflows/python-tests.yml/badge.svg)
```

**Review by garciparedes — CHANGES_REQUESTED** [review]
Could you apply the the proposed change?

File: docs/usage.rst [hunk 11]
```diff
@@ -2,6 +2,7 @@
 Usage
 =====
 
-To use Minos Microservice SAGA in a project::
+To use Minos Microservice Common in a project::
```
Comment (garciparedes) [inline docs/usage.rst:5]:
```suggestion
To use Minos Microservice Saga in a project::
```

File: docs/conf.py [hunk 12]
```diff
@@ -83,13 +91,24 @@
...
 # documentation.
 #
-# html_theme_options = {}
+
+# html_theme_options = {
+#     "codecov_button": True,
+#     "description": "Reactive microservices for an asynchronous world",
+#     "github_button": True,
+#     "github_user": "Clariteia",
+#     "github_repo": "minos_microservice_common",
```
Comment (garciparedes) [inline docs/conf.py:107]:
```suggestion
#     "github_repo": "minos_microservice_saga",
```

**Review by garciparedes — APPROVED** [review]
Hi @albamig, good changes! I'm sure that these documentation improvements will really help us to leverage this project to the next level! 🚀

**Comment by albamig** [comment]
Thank you for the thorough review @garciparedes :mag:

**Pull request merged**